src/Form/FormFieldControl.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useFormFieldContext } from './FormFieldContext';
+import { FormControl } from '..';
+import { useHasValue, callAllHandlers } from './fieldUtils';
+
+const FormFieldControl = React.forwardRef(({ className, ...props }, ref) => {
+  const {
+    size,
+    id,
+    isInvalid,
+    isValid,
+    onFocus,
+    onBlur,
+    placeholder,
+    'aria-describedby': ariaDescribedBy,
+  } = useFormFieldContext(props);
+
+  const [hasValue, checkInputEventValue] = useHasValue({ defaultValue: props.defaultValue, value: props.value });
+
+  return (
+    <FormControl
+      {...props}
+      ref={ref}
+      size={size}
+      id={id}
+      isInvalid={isInvalid}
+      isValid={isValid}
+      className={classNames(className, {
+        'has-value': hasValue,
+      })}
+      onFocus={onFocus}
+      /*
+        Force a whitespace string for a placeholder if none is defined so we can
+        use the :placeholder-shown pseudo-class when styling floating labels
+      */
+      placeholder={placeholder || ' '}
```

Comment:
Will this whitespace string placeholder affect screen readers in a negative way?